draft-ietf-quic-qpack.md

```diff
@@ -101,19 +101,238 @@ balance between resilience against head-of-line blocking and optimal compression
+header block might reference an entry in the dynamic table that has not yet been
+received.
+
+Each header block contains a Largest Reference {{header-prefix}} which
+identifies the table state necessary for decoding. If the greatest absolute
+index in the dynamic table is less than the value of the Largest Reference, the
+stream is considered "blocked."  While blocked, header field data should remain
+in the blocked stream's flow control window.  When the Largest Reference is
+zero, the frame contains no references to the dynamic table and can always be
+processed immediately. A stream becomes unblocked when the greatest absolute
+index in the dynamic table becomes greater than or equal to the Largest
+Reference for all header blocks the decoder has started reading from the stream.
+If a decoder encounters a header block where the actual largest reference is not
+equal to the Largest Reference declared in the prefix, it MAY treat this as a
+stream error of type HTTP_QPACK_DECOMPRESSION_FAILED.
+
+A decoder can permit the possibility of blocked streams by setting
+SETTINGS_QPACK_BLOCKED_STREAMS to a non-zero value (see {{configuration}}).
+This setting specifies an upper bound on the number of streams which can be
+blocked.
+
+An encoder can decide whether to risk having a stream become blocked. If
+permitted by the value of SETTINGS_QPACK_BLOCKED_STREAMS, compression efficiency
+can be improved by referencing dynamic table entries that are still in transit,
+but if there is loss or reordering the stream can become blocked at the decoder.
+An encoder avoids the risk of blocking by only referencing dynamic table entries
+which have been acknowledged, but this means using literals. Since literals make
+the header block larger, this can result in the encoder becoming blocked on
+congestion or flow control limits.
+
+An encoder MUST limit the number of streams which could become blocked to the
+value of SETTINGS_QPACK_BLOCKED_STREAMS at all times. Note that the decoder
+might not actually become blocked on every stream which risks becoming blocked.
+If the decoder encounters more blocked streams than it promised to support, it
+MUST treat this as a stream error of type HTTP_QPACK_DECOMPRESSION_FAILED.
+
+### Largest Known Received
+
+For the encoder to identify which dynamic table entries can be safely used
+without a stream becoming blocked, the encoder tracks the absolute index of the
+decoder's Largest Known Received entry.
+
+When blocking references are permitted, the encoder uses acknowledgement of
+header blocks to identify the Largest Known Received index, as described in
+{{header-acknowledgement}}.
+
+To acknowledge dynamic table entries which are not referenced by header blocks,
+for example because the encoder or the decoder have chosen not to risk blocked
+streams, the decoder sends a Table State Synchronize instruction (see
+{{table-state-synchronize}}).
+
+
+### Speculative table updates {#speculative-updates}
```

Comment:
I feel like this section doesn't qualify to be a separate section.  This is inherent in the fact that table updates and encoding a header set are separable operations.  I'd perhaps condense this to a single sentence and mention it in 2.1?

> Updates to the dynamic table can be performed independent of any particular header set being compressed.  Encoders MAY insert entries speculatively.

Title needs to be capitalized if retained.